src/unix/tcp.c

```diff
@@ -232,12 +232,12 @@ static int uv__ipv6_link_local_scope_id(void) {
   if (getifaddrs(&ifa))
     return 0;
 
-  for (p = ifa; p != NULL; p = p->ifa_next)
+  for (p = ifa; p != NULL && p->ifa_addr != NULL; p = p->ifa_next)
     if (uv__is_ipv6_link_local(p->ifa_addr))
       break;
```

Comment:
```suggestion
  for (p = ifa; p != NULL; p = p->ifa_next)
    if (p->ifa_addr != NULL)
      if (uv__is_ipv6_link_local(p->ifa_addr))
        break;
```
Otherwise the loop bails out too early. That also obviates the null check below.

Reply:
Ah, good catch. I'll fix that up.